tests/server/test_plugins.py

```diff
@@ -486,3 +491,464 @@ def test_history_of_collect_new_result(self):
+        worker.run()
+        assert 'status' in worker.result and worker.result['status'] == 'OK'
+        assert 'info' in worker.result['messages']
+        assert worker.result['messages']['info'] == [os.uname()[1]]
+
+    def test_running_worker_process_title_should_be_changed(self):
+        expected = 'smokerd plugin Hostname'
+        worker = server_plugins.PluginWorker(**self.conf_worker)
+        worker.run()
+        procs = get_process_list()
+        assert expected in procs.values()
+
+    def test_drop_privileged_with_invalid_params(self):
+        test_params = {
+            'uid': 99999,
+            'gid': 99999
+        }
+        params = dict(self.params_default, **test_params)
+        worker = server_plugins.PluginWorker(name='Hostname',
+                                             queue=self.queue, params=params)
+        with pytest.raises(OSError) as exc_info:
+            worker.run()
+        assert 'Operation not permitted' in exc_info.value
+
+    def test_drop_privileged_with_invalid_params_type(self):
+        test_params = {
+            'uid': 'InvalidType',
+            'gid': 'InvalidType',
+        }
+        params = dict(self.params_default, **test_params)
+        worker = server_plugins.PluginWorker(name='Hostname',
+                                             queue=self.queue, params=params)
+        with pytest.raises(TypeError) as exc_info:
+            worker.run()
+        assert 'an integer is required' in exc_info.value
+
+    def test_run_worker_with_maintenance_lock(self):
+        expected_message = ['Skipped because of maintenance in progress']
+
+        maintenance_lock = os.getcwd() + random_string()
+        test_params = {'MaintenanceLock': maintenance_lock + '.lock'}
+        params = dict(self.params_default, **test_params)
+        lock = lockfile.FileLock(maintenance_lock)
+        with lock:
+            worker = server_plugins.PluginWorker(name='Hostname',
+                                                 queue=self.queue,
+                                                 params=params)
+            worker.run()
+            assert 'status' in worker.result
+            assert worker.result['status'] == 'WARN'
+            assert 'warn' in worker.result['messages']
+            assert worker.result['messages']['warn'] == expected_message
+
+    def test_run_invalid_command(self):
+        expected_message = ['/bin/sh: 1: InvalidCommand: not found']
+        test_params = {
+            'Command': 'InvalidCommand'
+        }
+        params = dict(self.params_default, **test_params)
+        worker = server_plugins.PluginWorker(name='InvalidCommand',
+                                             queue=self.queue, params=params)
+        worker.run()
+        assert 'status' in worker.result and worker.result['status'] == 'ERROR'
+        assert 'warn' in worker.result['messages']
+        assert worker.result['messages']['error'] == expected_message
```

Comment:
Fails on my machine... Probably related to bash version or bash configuration. Changing to regexp could probably solve this. 
```bash
_________________________________________________ TestPluginWorker.test_run_invalid_command _________________________________________________

self = <tests.server.test_plugins.TestPluginWorker object at 0x10216f110>

    def test_run_invalid_command(self):
        expected_message = ['/bin/sh: 1: InvalidCommand: not found']
        test_params = {
            'Command': 'InvalidCommand'
        }
        params = dict(self.params_default, **test_params)
        worker = server_plugins.PluginWorker(name='InvalidCommand',
                                             queue=self.queue, params=params)
        worker.run()
        assert 'status' in worker.result and worker.result['status'] == 'ERROR'
        assert 'warn' in worker.result['messages']
>       assert worker.result['messages']['error'] == expected_message
E       assert ['/bin/sh: In...nd not found'] == ['/bin/sh: 1: ...d: not found']
E         At index 0 diff: '/bin/sh: InvalidCommand: command not found' != '/bin/sh: 1: InvalidCommand: not found'
E         Use -v to get the full diff
```